### PR TITLE
SQL: [Tests] Add integ tests for selecting a literal and an aggregate

### PR DESCRIPTION
**Pull request by emasab**

### CODE REVIEW

**Comment by emasab** [comment]
Closes #41411

**Comment by tvernum** [comment]
@elasticmachine test this please

**Comment by emasab** [comment]
otherwise this query was failing because the literal now is supportedByAggsOnlyQuery

> SELECT DAY_NAME(birth_date + INTERVAL 1 DAY) FROM test_emp

**Comment by emasab** [comment]
the last commit fixes some integration tests that were failing, pending to add more integration tests, at the moment there is only one unit tests for the issue that is addressing

**Comment by tvernum** [comment]
@elastic/es-sql Can someone look at this community PR?

**Comment by emasab** [comment]
rebased on current master

**Comment by emasab** [comment]
rebased

**Comment by emasab** [comment]
rebased and some formatting

**Comment by emasab** [comment]
rebased

**Comment by matriv** [comment]
Hi @emasab 

Apologies for the delay here. Since the merging of #49570 seems that the issue is solved.
It would be great though to add the tests that you implemented to have this behaviour with
literals and aggregations tested thoroughly. Would you mind adjusting your PR to only add the 
new tests?

Thank you!

**Comment by emasab** [comment]
Hi @matriv, happy to hear from you! I'm going to do that asap.

Best

**Comment by matriv** [comment]
@emasab Thank you!

**Comment by emasab** [comment]
That's it! Only the tests are left and they're passing.

File: x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/planner/QueryFolderTests.java [hunk 1]
```diff
@@ -430,4 +430,17 @@ public void testFoldingOfPivot() {
         assertThat(a, containsString("\"terms\":{\"field\":\"keyword\""));
         assertThat(a, containsString("{\"avg\":{\"field\":\"int\"}"));
     }
+
+    public void testFoldingOfLiteralWithGroupBy() {
```
Comment (matriv) [inline x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/planner/QueryFolderTests.java]: You can remove this, as it's already added here: https://github.com/elastic/elasticsearch/blob/26283b63882e85ec7aca09cae8bd046f01bdd8c9/x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/planner/QueryFolderTests.java#L383

**Review by matriv — APPROVED** [review]
LGTM, left a comment regarding the unit test.

**Comment by matriv** [comment]
@astefan Could you please also take a look?

**Comment by emasab** [comment]
@matriv I've removed the duplicated unit test

**Review by astefan — APPROVED** [review]
LGTM

**Comment by matriv** [comment]
@elasticmachine ok to test

**Comment by matriv** [comment]
@emasab Thanks a lot for your patience and your contribution!

**Pull request merged**